### DISCUSSION TITLE
Ability to lock panel query to alert query

### DISCUSSION THREAD

**Discussion by gtirloni** [grafana/grafana#73235] [Alerting]
When a alert is created through the `Create alert rule from this panel` buttom, the alert query is populated with the panel's query initially.

Afterwards, any change to the alert's or panel's query don't affect each other.

This might create confusion for users migrated from Legacy to Unified Alerting and expecting some sort of dependency between them.

Grafana could offer the option to keep the queries synchronized (maybe by adding metadata to the query to link it to the panel's query somehow) and them grey out the alert's query text so users can't change (or inform users that changes in the alert's query would change the panel's query since they are synchronized).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82555. Feel free to continue the discussion around this there. Thank you!
